applications/serial_lte_modem/src/http_c/slm_at_httpc.c

```diff
@@ -618,48 +660,46 @@ int handle_at_httpc_request(enum at_cmd_type cmd_type)
 		return err;
 	}
 
-	if (httpc.state != HTTPC_INIT) {
-		LOG_ERR("Another request is not finished.");
-		return err;
-	}
-
 	switch (cmd_type) {
 	case AT_CMD_TYPE_SET_COMMAND:
 		param_count = at_params_valid_count_get(&at_param_list);
-		err = util_string_get(&at_param_list, 1,
-					   data_buf, &method_sz);
+		httpc.pl_len = 0;
+		httpc.total_sent = 0;
+		httpc.state = HTTPC_INIT;
+		memset(data_buf, 0, sizeof(data_buf));
+		err = util_string_get(&at_param_list, 1, data_buf, &method_sz);
 		if (err < 0) {
 			LOG_ERR("Fail to get method string: %d", err);
 			return err;
 		}
 		httpc.method_str = (char *)data_buf;
 		offset = method_sz + 1;
 		/* Get resource path string */
-		err = util_string_get(&at_param_list, 2,
-					   data_buf + offset, &resource_sz);
+		err = util_string_get(&at_param_list, 2, data_buf + offset, &resource_sz);
 		if (err < 0) {
 			LOG_ERR("Fail to get resource string: %d", err);
 			return err;
 		}
 		httpc.resource = (char *)(data_buf + offset);
 		offset = offset + resource_sz + 1;
 		/* Get header string */
-		err = util_string_get(&at_param_list, 3,
-					   data_buf + offset, &headers_sz);
+		err = util_string_get(&at_param_list, 3, data_buf + offset, &headers_sz);
 		if (err < 0) {
 			LOG_ERR("Fail to get option string: %d", err);
 			return err;
 		}
 		httpc.headers = (char *)(data_buf + offset);
 		if (param_count >= 5) {
-			err = at_params_int_get(&at_param_list, 4,
-						  &httpc.pl_len);
+			err = at_params_int_get(&at_param_list, 4, &httpc.pl_len);
 			if (err != 0) {
 				return err;
 			}
 		}
-		/* start sending request */
-		k_sem_give(&http_req_sem);
+		/* start http request thread */
+		k_thread_create(&httpc_thread, httpc_thread_stack,
+			K_THREAD_STACK_SIZEOF(httpc_thread_stack),
+			httpc_thread_fn, NULL, NULL, NULL,
+			HTTPC_THREAD_PRIORITY, K_USER, K_NO_WAIT);
```

Comment:
nit: alignment
```suggestion
		k_thread_create(&httpc_thread, httpc_thread_stack,
				K_THREAD_STACK_SIZEOF(httpc_thread_stack),
				httpc_thread_fn, NULL, NULL, NULL,
				HTTPC_THREAD_PRIORITY, K_USER, K_NO_WAIT);
```

Reply:
The code was changed per suggestion, thank you.